backend/src/main/controller/PromotionController.ts

```diff
@@ -13,13 +13,17 @@ import {
   PromotionQueryDTO,
   PromotionQueryValidation,
 } from '../validation/PromotionQueryValidation';
+import { CachingService } from '../service/CachingService';
+import { CachingObject } from '../data/CachingObject';
 
 export class PromotionController {
+  private cachingService = new CachingService();
```

Comment:
That could work! Where were you thinking of instantiating it then?

Reply:
Right now we instantiate all our dependencies inside `App.ts`. In the future, I am thinking we can use Inversion of Control containers for all our service classes since the service classes need to be stateless.

Example: https://www.npmjs.com/package/inversify

Reply:
Gotcha! That makes sense, I'll check the link out and change it!